advanced_topics.qmd

```diff
@@ -5,13 +5,114 @@
+
+So how do we use this from the arrow R package?
+We start off by considering which R function to use.
+Let's go with the base R function `startsWith()`.
+The table below shows a comparison with the Acero equivalent.
+
+| Feature                     | R (`startsWith()`)                                | Acero (`starts_with()`)                            |
+|-----------------|------------------------|--------------------------------|
+| **Function Name**           | `startsWith`                                      | `starts_with`                                      |
+| **Function Arguments**      | `prefix`                                     | `pattern`, `ignore_case = false`              |
+| **Purpose**                 | Checks if a string starts with a specified prefix | Checks if a string starts with a specified pattern |
+| **Prefix/Pattern Argument** | `prefix`: The prefix to match                     | `pattern`: The pattern to match                    |
+| **Case Sensitivity**        | Always case-sensitive                             | `ignore_case`: Boolean, set to `false` by default  |
+
+There is already a mapping between these two functions, and we can see this by creating our query but not actually executing it.
+
+```{r}
+#| label: startsWith
+open_dataset("data/person/") |> 
+  select(location) |> 
+  filter(startsWith(x = location, prefix = "w"))
+```
+
+The information about the Acero mapping is shown here in the line which starts with "\* Filter" - this contains a human-readable representation of the Arrow Expression created.
+
+It shows that that `startsWith()` has been translated to the Acero function `starts_with()`.
+The options for the Acero function have been set based on the values which were passed in via the R expression.
+We set the `startsWith()` argument `prefix` to "w", which was passed to the Acero `starts_with()` option called `pattern`.
+As `startsWith()` is always case-sensitive, the Acero `starts_with()` option `ignore_case` has been set to `false`.
+
+This approach is great for the most typical case where we know the R function we want to use as arrow automatically translates to the relevant Acero function, but what about if we just want to start with Acero functions?
+
+We can view the full list of Acero functions available to use in arrow by calling `list_compute_functions()`.
+At the time of writing there were 262 Acero functions available to use in R - the first 20 are shown below.
+
+```{r}
+#| label: list_compute_functions
+list_compute_functions() |> 
+  head(20)
+```
+
+You can also use these functions in your arrow dplyr pipelines by prefixing them with `"arrow_"`.
+
+This isn't necessary when working with Acero functions which have a direct mapping to their R equivalents - which can be used instead. 
+However, there are some niche use-cases where this could be helpful - for example, if you want to use Acero options which don't exist in the R equivalent function.
```

Comment:
This is true, but it would be more compelling of an example if it actually used an option that wasn't available through a regular binding.

Reply:
Good point!